### PR TITLE
fix(addon-mobile): close sheet when extended to full state

### LINKED ISSUE #2057: 🐞 -  При нажатии на затемненную область вне Sheet компонента, он не закрывается
### Which @taiga-ui/* package(s) are the source of the bug?

addon-mobile

### Please provide a link to a minimal reproduction of the bug

_No response_

### Is this issue blocking you?

Non-Blocking

### Description

Возспроизвели в проприетарной версии в навигации.

### Angular version

11.2.14

### Taiga UI version

2.51

### Which browsers have you used?

- [X] Chrome
- [X] Firefox
- [X] Safari
- [ ] Edge

### Which operating systems have you used?

- [X] macOS
- [ ] Windows
- [ ] Linux
- [X] iOS
- [X] Android

### PR DESCRIPTION
**Pull request by splincode**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?

Closes #2057

### CODE REVIEW

File: projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.component.ts [hunk 1]
```diff
@@ -22,6 +22,14 @@ export class TuiSheetsHostComponent {
         @Inject(TUI_WINDOW_HEIGHT) readonly height$: Observable<number>,
     ) {}
 
+    closeIfOutside(event: Event, item: TuiSheet<unknown>): void {
+        const isOutside = !(event.target as HTMLElement)?.closest('tui-sheet');
```
Comment (nsbarsukov) [inline projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.component.ts:26]:
```ts
@HostListener('click', ['$event.target'])
click(target: HTMLElement): void {
    const isOutside = target.closest('tui-sheet');

    if (isOutside) {
        this.close(item);
    }
}
```

Comment (nsbarsukov) [reply projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.component.ts:26]: oops, my mistake =)

**Review by nsbarsukov — APPROVED** [review]

File: projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.template.html [hunk 2]
```diff
@@ -7,6 +7,7 @@
         [tuiSheetWrapper]="item.offset"
         [class.t-wrapper_overlay]="item.overlay"
         [class.t-wrapper_closeable]="item.closeable"
+        (click)="closeIfOutside($event, item)"
```
[the remark this answers is not in the document]
Comment (splincode) [reply projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.template.html]: fixed

[the remark this answers is not in the document]
Comment (waterplea) [reply projects/addon-mobile/components/sheet/components/sheets-host/sheets-host.template.html]: I said it didn't work. For some reason when you click on this label here, the target is the wrapper. Didn't have time to investigate:
image

**Review by waterplea — APPROVED** [review]

**Pull request merged**